source/module_base/test/complexarray_test.cpp

```diff
@@ -45,125 +50,417 @@
 *   - operator "*=":
 *       similar as "*"
 *
+*   - operator "==":
+*       judge if two ComplexArray is equal, all element is equal.
+*
+*   - operator "!=":
+*       judge if two ComplexArray is not equal
+*
 *   - oprator "()":
 *       - access the element
+*
+*   - function zero_out, negate, randomize, getBound/Size, create.
+*
+*   Test relative functions:
+*   - overloading of operator "*". a double/complex multiply a Class ComplexArray.
+*   - functon abs2(), return the sum of squares of all elements 
+*   - function dot()
+*   - function scale_accumulate()
+*   - function scaled_sum()
+*   - fucntion point_mult()
 */
 
 
+double prec_test = 1e-14;
+//compare two complex by using EXPECT_NEAR()
+void EXPECT_COMPLEXD(const std::complex<double>& a,const std::complex<double>& b)
```

Comment:
The name `EXPECT_COMPLEXD` should be more distinct. Maybe something like `EXPECT_COMPLEX_DOUBLE_NEAR` or `EXPECT_COMPLEX_EQUAL`, indicating type ("complex", "complex<double>") and method("NEAR", "EQ").